### PR TITLE
bug fix and additional testing on cyl mesh

### LINKED ISSUE #367: Problem with aveE2CCV when used on a cylindrical mesh
Use of the aveE2CCV function reduces values by a factor of 0.5 when used on a cylindrical mesh. 

Below is a simple example which shows this behaviour.

`# import unittest
from SimPEG import EM, Mesh, Utils, np, Maps, sp
# import sys
from scipy.constants import mu_0
%pylab inline
import matplotlib.pyplot as plt

# Create cylindrical mesh
csx, ncx, npadx = 5, 50, 25
csz, ncz, npadz = 5, 50, 25
hx = Utils.meshTensor([(csx, ncx), (csx, npadx, 1.3)])
hz = Utils.meshTensor([(csz, npadz, -1.3), (csz, ncz), (csz, npadz, 1.3)])
mesh = Mesh.CylMesh([hx, 1, hz], [0., 0., -hz.sum()/2])

# Dummy H function which lives on edges
h_e = np.ones(mesh.nEy)

# average to CC
h_cc = mesh.aveE2CCV*h_e

# Measurement locations for comparison
rlim = [20., 500.]
r = mesh.vectorCCx[np.argmin(np.abs(mesh.vectorCCx-rlim[0])):np.argmin(np.abs(mesh.vectorCCx-rlim[1]))]
z = 100.
XYZ = Utils.ndgrid(r, np.r_[0.], np.r_[z])

# Create data projection matrices
Pcc = mesh.getInterpolationMat(XYZ, 'CC')
Pey = mesh.getInterpolationMat(XYZ, 'Ey')

h_cc_test = Pcc*h_cc
h_e_test = Pey*h_e

# Plot
plt.subplot(211)
plt.plot(r,h_cc_test.real,'o',r,h_e_test.real,linewidth=2)
plt.grid(which='both')
plt.title('H_y Real')
plt.xlabel('r (m)')

plt.subplot(212)
plt.plot(r,h_cc_test.imag,'o',r,h_e_test.imag,linewidth=2)
plt.grid(which='both')
plt.title('H_y Imag')
plt.legend(['Num','Ana'],bbox_to_anchor=(1.5,0.5))
plt.xlabel('r (m)')

plt.tight_layout()
`

image

### PR DESCRIPTION
**Pull request by lheagy**
- bug fix in `aveE2CC` and `aveE2CCV` that was dividing everything by 2 to close #367 
- testing on `aveE2CC`, `aveE2CCV`, `aveF2CC`, `aveF2CCV`

@micmitch : can you take a look at this? If you have any questions / anything is unclear, please let me know!

### CODE REVIEW

File: SimPEG/Mesh/CylMesh.py [hunk 1]
```diff
@@ -1,12 +1,11 @@
 import numpy as np
 import scipy.sparse as sp
 from scipy.constants import pi
-from SimPEG.Utils import mkvc, ndgrid, sdiag, kron3, speye, spzeros, ddx, av, avExtrap
+from SimPEG import Utils
```
Comment (lheagy) [inline SimPEG/Mesh/CylMesh.py]: sorry, this makes this not an overly transparent pr, but is a helpful clean-up for code readability. I will highlight the important aspects with comments

File: SimPEG/Mesh/CylMesh.py [hunk 2]
```diff
@@ -322,14 +321,113 @@ def aveF2CCV(self):
...
+            M = self.dim * Utils.sdiag(Av.T * Vprop)
+        elif prop.size == self.nC*self.dim:
+            Av = getattr(self, 'ave'+projType+'2CCV')
+            V = sp.kron(sp.identity(self.dim), Utils.sdiag(self.vol))
+            M = Utils.sdiag(Av.T * V * Utils.mkvc(prop))
+        else:
+            return None
+
+        if projType == 'E':
+                M = 0.5*M
```
Comment (lheagy) [inline SimPEG/Mesh/CylMesh.py]: for the edge inner products to function properly (by this I mean, have the EM analytics pass), we need this factor of 0.5 due to the bug fix in the averaging. It is not obvious to me why we need this - @rowanc1 : can you shed some light here?

File: SimPEG/Mesh/CylMesh.py [hunk 3]
```diff
@@ -277,14 +276,14 @@ def aveE2CC(self):
             # The number of cell centers in each direction
             n = self.vnC
             if self.isSymmetric:
-                avR = av(n[0])[:,1:]
+                avR = Utils.av(n[0])[:, 1:]
                 avR[0,0] = 1.
-                self._aveE2CC = (0.5)*sp.kron(av(n[2]), avR, format="csr")
+                self._aveE2CC = sp.kron(Utils.av(n[2]), avR, format="csr")
```
Comment (lheagy) [inline SimPEG/Mesh/CylMesh.py]: this is where the factor of 2 bug was.

File: tests/mesh/test_cylMesh.py [hunk 4]
```diff
@@ -356,6 +356,116 @@ def test_order(self):
...
+
+    def setUp(self):
+        hx = np.random.rand(10.)
+        hz = np.random.rand(10.)
+        self.mesh = Mesh.CylMesh([hx, 1,hz])
+
+    def test_constantEdges(self):
+        edge_vec = np.ones(self.mesh.nE)
+        assert all(self.mesh.aveE2CC * edge_vec == 1.)
+        assert all(self.mesh.aveE2CCV * edge_vec == 1.)
```
Comment (rowanc1) [inline tests/mesh/test_cylMesh.py]: seems like we should have caught this earlier. :)

**Pull request merged**